### PR TITLE
goose down [version_number]

### PR DESCRIPTION
**Pull request by mikepigott**
https://www.pivotaltracker.com/projects/1110448/stories/112433943

Add an optional argument to `goose down` to allow it to rollback to a specific version.

Also, given two directories of database migration scripts:
(1) Find out what the latest common version number is.
(2) `goose down` to that version.

### CODE REVIEW

**Comment by mikepigott** [comment]
https://www.pivotaltracker.com/projects/1110448/stories/112433943

File: lib/goose/migrate.go [hunk 1]
```diff
@@ -339,6 +339,75 @@ func GetMostRecentDBVersion(dirpath string) (version int64, err error) {
...
+
+	sort.Ints(firstVersions)
+	sort.Ints(secondVersions)
+
+	// Assumes both directories have the same first version number.
+	fvLen := len(firstVersions)
+	svLen := len(secondVersions)
+
+	maxLen := fvLen
+	downTo := firstVersions[len(firstVersions) - 1]
```
Comment (crro) [inline lib/goose/migrate.go]: You could use fvLen since you declared it above

File: lib/goose/migrate.go [hunk 2]
```diff
@@ -339,6 +339,75 @@ func GetMostRecentDBVersion(dirpath string) (version int64, err error) {
...
+
+	if svLen < fvLen {
+		maxLen = svLen
+		downTo = secondVersions[len(secondVersions) - 1]
```
Comment (crro) [inline lib/goose/migrate.go]: same here with svLen

**Comment by crro** [comment]
Other than two small suggestions that you may leave as is, LGTM

**Pull request merged**